bin/build-helmet.js

```diff
@@ -1,42 +1,68 @@
 #!/usr/bin/env node
+import { promises as fs } from "fs";
 import * as path from "path";
 import { fileURLToPath } from "url";
 import rollupTypescript from "@rollup/plugin-typescript";
-import { writeRollup, withCommonJsFile } from "./helpers.js";
+import {
+  writeRollup,
+  withCommonJsFile,
+  withEsmFile,
+  renameFile,
+  finalizeCommonJs,
+  moveDir
+} from "./helpers.js";
 
 const thisPath = fileURLToPath(import.meta.url);
 const rootPath = path.join(path.dirname(thisPath), "..");
+const distPath = path.join(rootPath, "dist");
 const esmSourcePath = path.join(rootPath, "index.ts");
-const esmDistPath = path.join(rootPath, "dist", "index.js");
-const commonJsDistPath = path.join(rootPath, "dist", "index.cjs");
+const esmDistDir = path.join(distPath, "esm");
+const esmDistPath = path.join(esmDistDir, "index.js");
+const commonJsDistDir = path.join(distPath, "cjs");
+const commonJsDistPath = path.join(commonJsDistDir, "index.js");
+const typesDistDir = path.join(distPath, "types");
 
 const compileEsm = () =>
-  writeRollup(
-    {
-      input: esmSourcePath,
-      plugins: [rollupTypescript({ tsconfig: "./tsconfig-esm.json" })],
-    },
-    { file: esmDistPath }
+  withEsmFile(esmSourcePath, (esmTempPath) =>
+    writeRollup(
+      {
+        input: esmTempPath,
+        plugins: [rollupTypescript({ tsconfig: "./tsconfig-esm.json" })],
+      },
+      { file: esmDistPath }
+    )
   );
 
 const compileCommonjs = () =>
-  withCommonJsFile(esmSourcePath, (commonJsSourcePath) =>
+  withCommonJsFile(esmSourcePath, (commonJsTempPath) =>
     writeRollup(
       {
-        input: commonJsSourcePath,
+        input: commonJsTempPath,
         plugins: [rollupTypescript({ tsconfig: "./tsconfig-commonjs.json" })],
       },
       {
-        exports: "default",
+        exports: "named",
         file: commonJsDistPath,
         format: "cjs",
       }
-    )
+    ).then(async () => await finalizeCommonJs(commonJsDistDir, esmDistDir))
+  );
+
+const finalizeTypes = async () => {
+  await fs.mkdir(typesDistDir);
+  await fs.mkdir(path.join(typesDistDir, "middlewares"));
```

Comment:
Nitpick: could we shorten this to one operation?

```suggestion
  await fs.mkdir(path.join(typesDistDir, "middlewares"), { recursive: true });
```